modules/edu.gemini.seqexec.web/edu.gemini.seqexec.web.client/src/main/scala/edu/gemini/seqexec/web/client/model/SeqexecCircuit.scala

```diff
@@ -232,7 +233,7 @@ class WebSocketHandler[M](modelRW: ModelRW[M, WebSocketConnection]) extends Acti
     def onMessage(e: MessageEvent): Unit = {
       val byteBuffer = TypedArrayBuffer.wrap(e.data.asInstanceOf[ArrayBuffer])
       \/.fromTryCatchNonFatal(Unpickle[SeqexecEvent].fromBytes(byteBuffer)) match {
-        case \/-(event) => println(s"Decoding event $event") // SeqexecCircuit.dispatch(NewSeqexecEvent(event))
+        case \/-(event) => SeqexecCircuit.dispatch(NewSeqexecEvent(event))
```

Comment:
The `sealed` comes as scalacheck-shapeless cannot derive an arbitrary if it doesn't know all the possible implementations. You can use normal traits in many cases, e.g. when defining a simple interface